### ISSUE TITLE
Weird build issue producing binary that hangs early on startup

### ISSUE THREAD

**Issue by bubbleguuum** [atlas-engineer/nyxt#2624]
Here's a weird one.

Building current nyxt git on openSUSE Tumbleweed using SBCL 2.2.10 with **make all**, the produced nyxt binary 
hangs early on startup. See strace, it seems to mysteriously hang on a futex() call.
[nyxt_strace.txt](https://github.com/atlas-engineer/nyxt/files/9945807/nyxt_strace.txt)

After trying many things, including compiling with  previous SBCL versions (2.2.9 and 2.2.8, the latter I'm certain it worked at some point), this issue was still there.

When I was going to give up,  totally mystified about this issue, I accidentally found a workaround that fixes it:

```
make all
rm nyxt
make all
```

First **make all** compile the dependencies and build the nyxt binary.
The remaining 2 commands force only to rebuild the nyxt binary. For some reason it works, and the resulting 
binary launches fine.
 
So there's some weird issue causing a faulty binary when just **make all** is invoked to build deps + final binary.
It would be interesting if someone can reproduce this issue on his system.

**Comment by Ambrevar** [comment]
Someone reported something similar some years ago, but the problem on the user side, if I recall correctly.

Can you double-check that the first  `make all` is indeed producing a new `nyxt` binary, instead of being a no-op?
In particular, check the date of the file (e.g. with `ls -l`).

**Comment by bubbleguuum** [comment]
first make all does produce a binary:

```
~/git_projects/nyxt> ls -l nyxt
-rwxr-xr-x 1 bobbie users 152398008 Nov  7 11:34 nyxt
```

As per first post, starting it locks up early on startup:

```
~/git_projects/nyxt> ./nyxt
Nyxt version 3-pre-release-2-227-ge853e0f4
<INFO> [11:40:09] Source location: #P"/home/bobbie/git_projects/nyxt/"
```
Althought `nyxt --version` and `nyxt --system-information` work fine.

after removing the binary and just rebuilding it with make all, it magically works.
The binary is smaller:

```
~/git_projects/nyxt> ls -l nyxt
-rwxr-xr-x 1 bobbie users 149284248 Nov  7 11:43 nyxt
```

Looking at the end of build traces, I can see some things are different. I've included the end of 
both build log for comparison.

[build_not_working.log](https://github.com/atlas-engineer/nyxt/files/9950710/build_not_working.log)
[build_working.log](https://github.com/atlas-engineer/nyxt/files/9950711/build_working.log)

**Comment by Ambrevar** [comment]
@bubbleguuum Can you test #2654 see if it fixes it for you?

**Comment by bubbleguuum** [comment]
I tried it and it does not fix that issue.

The mystery deepens as a `make all` from a fresh 'git clone`, _sometimes_ produces a `nyxt` binary that works.
I did that 3 times (clone followed by `make all`): the first 2 times it produced a working binary, the third time it didn't.

However,  after a build producing a `nyxt` binary (working or not), this sequence of commands will always produce a non-working binary (even with the PR mentioned applied):

```
make clean 
rm nyxt
make all
``` 

There is something funky going on with the build, at least on my system.
The only reliable way to build a working binary is `make all; rm nyxt; make all`. 
It produces a working binary 3MB smaller than a single `make all`, which is strange, but probably pointing
at the Lisp image not containing exactly the same stuff.
